Display Edit notices if available

staykids opened the pull request:
**Phabricator:** https://phabricator.wikimedia.org/T201596

### Notes
This adds the presentation of the Edit notices modal (if Edit notices are available) when the user edits an article. 

Some outstanding items:
- While in my testing so far on device it's uncommon that the section editor's keyboard toolbar overlaps the modal that's presented, I think this is still a corner case we may need to address
- Still need to connect to the section editor dismiss confirmation action sheet when tapping a URL

### Test Steps
It's best to start from a delete/fresh install to confirm the UserDefaults for the editing welcome controller are reset.

1. Find an article with Edit notices (I often use `Covid-19`)
2. Tap to edit a section and confirm you see the editing welcome controller
3. After completing the editing welcome, confirm you are presented with the section editor and a modal appears with any edit notices (note, you can also alter `EditNoticesFetcher` to remove the filtering by `editnotice` to see all available notices displayed here).
4. Toggle the switch and return to step 1. Confirm editing again does not automatically display the article, but adds a bar button to display the modal again.
5. Try changing the app's theme and confirm the modal still renders as expected.
6. Tap a link in the modal and confirm it dismisses the modal and editor then pushes the URL over the article view controller.

tonisevener reviewed (commented):
Looking good! Thanks for making the PR easy to review. Just a couple of code comments and this design comment about this Done button dipping into the safe area:

![Screen Shot 2023-02-08 at 1 45 48 PM](https://user-images.githubusercontent.com/3620196/217636100-b43e4c55-a23b-4cf0-b952-3ea9e3b9137c.png)

I did notice that keyboard toolbar issue you mentioned a couple of times, but I don't think it makes sense for us to investigate until both this and blocked messages are merged.

staykids commented on Wikipedia/Code/EditNoticesViewController.swift:
```diff
@@ -0,0 +1,77 @@
+    func editNoticesControllerUserTapped(url: URL)
+}
+
+class EditNoticesViewController: ViewController {
```
Mostly just did this out of inertia and to minimize having to add boilerplate surrounding initialization and theme management to the controller. If it still makes you uncomfortable though, let me know and I can refactor it.

tonisevener replied:
Yeah if you don't mind - we have a `ThemeableViewController` that you could subclass, it might save you a little bit of code on theming. You'll be able to remove lines 33-34 as well. Those are probably harmless but would be nice to remove the `scrollView` assignment if it won't be used. After some tinkering looks like this init change works:

```
    init(theme: Theme, viewModel: EditNoticesViewModel) {
        self.viewModel = viewModel
        super.init(nibName: nil, bundle: nil)
        self.theme = theme
    }
```

staykids replied:
> Yeah if you don't mind - we have a `ThemeableViewController` that you could subclass

@tonisevener No problem – done!

staykids commented on Wikipedia/Code/SectionEditorViewController.swift:
```diff
@@ -342,15 +389,17 @@ class SectionEditorViewController: ViewController {
                 case .failure(let error):
                     self.didFocusWebViewCompletion = {
                         WMFAlertManager.sharedInstance.showErrorAlert(error as NSError, sticky: true, dismissPreviousAlerts: true)
+                        self.initialFetchGroup.leave()
```
Good catch – updated!

staykids commented:
@tonisevener Thanks for the review notes – I think this is ready for another look:
- Moved task group `leave()` call
- Updated done button trailing constraint to follow readable content guide while still respecting design-desired margin
- Updated URL navigation away from edit notices modal to go through confirm discard edits alert. Also refactored the name of that alert function. It's not super elegant the way I crammed the post-dismiss URL to present into the delegate method, but it works. Happy to refactor it if you'd prefer an alternate pattern.
